NavigationRail/src/main/java/com/microsoft/device/display/samples/navigationrail/ui/view/GalleryView.kt

```diff
@@ -5,31 +5,117 @@
 import androidx.compose.foundation.Image
 import androidx.compose.foundation.border
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.lazy.GridCells
 import androidx.compose.foundation.lazy.LazyVerticalGrid
 import androidx.compose.foundation.lazy.items
 import androidx.compose.foundation.selection.selectable
 import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Scaffold
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
 import com.microsoft.device.display.samples.navigationrail.R
 import com.microsoft.device.display.samples.navigationrail.models.Image
+import com.microsoft.device.display.samples.navigationrail.ui.components.GalleryBottomNav
+import com.microsoft.device.display.samples.navigationrail.ui.components.GalleryNavRail
+import com.microsoft.device.display.samples.navigationrail.ui.components.GalleryTopBar
+import com.microsoft.device.dualscreen.twopanelayout.Screen
+import com.microsoft.device.dualscreen.twopanelayout.TwoPaneNavScope
 
 private val BORDER_SIZE = 7.dp
 private val GALLERY_SPACING = 2.dp
 private const val NUM_COLUMNS = 2
 
+@ExperimentalAnimationApi
+@ExperimentalFoundationApi
+@Composable
+fun TwoPaneNavScope.GalleryViewWithTopBar(
+    section: GallerySections,
+    horizontalPadding: Dp,
+    navController: NavHostController,
+    isDualScreen: Boolean,
+    imageId: Int?,
+    updateImageId: (Int?) -> Unit,
+    currentRoute: String,
+    updateRoute: (String) -> Unit
+) {
+    // Make sure the current gallery section should be displayed
+    DisposableEffect(isSinglePane) {
+        if (isSinglePane && imageId != null) {
+            navController.navigateTo(ITEM_DETAIL_ROUTE, Screen.Pane2, defaultNavOptions)
+        } else if (section.route != currentRoute) {
+            navController.navigateTo(currentRoute, screen = Screen.Pane1, defaultNavOptions)
+        }
+
+        // REVISIT: check if another effect would fit this use case better
+        onDispose { }
+    }
+
+    BackHandler(enabled = isSinglePane && navController.backQueue.size > 2) {
+        navController.previousBackStackEntry?.destination?.route?.let { updateRoute(it) }
+        navController.navigateUp()
+        logBackQueue(navController)
+    }
+
+    val twoPaneNavScope = this
+
+    // Use navigation rail when dual screen (more space), otherwise use bottom navigation
+    Scaffold(
+        bottomBar = {
+            if (!isDualScreen)
```

Comment:
I think we use `isSinglePane` here as above.

Reply:
For these, I think we should leave them as `isDualScreen` only because we're using `HorizontalSingle` mode, so `isSinglePane` will return false in dual landscape mode (but it's still an expanded screen size so we would still want to use a navigation rail) - does that sound ok?